src/Adapter/Product/Update/ProductSupplierUpdater.php

```diff
@@ -79,15 +78,68 @@ public function __construct(
     /**
      * @param ProductId $productId
      * @param SupplierId $defaultSupplierId
-     * @param ProductSupplier[] $productSuppliers
+     * @param array<int, ProductSupplier> $productSuppliers
      *
-     * @throws CannotUpdateProductSupplierException
-     * @throws CoreException
-     * @throws ProductSupplierNotFoundException
+     * @return array<int, ProductSupplierId>
+     */
+    public function setProductSuppliers(
+        ProductId $productId,
+        SupplierId $defaultSupplierId,
+        array $productSuppliers
+    ): array {
+        $this->persist($productId, $productSuppliers);
+        $this->updateDefaultSupplier($productId, $defaultSupplierId);
+
+        return $this->getCurrentProductSupplierIds($productId);
+    }
+
+    /**
+     * @param ProductId $productId
+     * @param CombinationId $combinationId
+     * @param array<int, ProductSupplier> $productSuppliers
+     *
+     * @return array<int, ProductSupplierId>
      */
-    public function setProductSuppliers(ProductId $productId, SupplierId $defaultSupplierId, array $productSuppliers): void
+    public function setCombinationSuppliers(
+        ProductId $productId,
+        CombinationId $combinationId,
+        array $productSuppliers
+    ): array {
+        $this->persist($productId, $productSuppliers, $combinationId);
+
+        return $this->getCurrentProductSupplierIds($productId, $combinationId);
+    }
+
+    /**
+     * Removes associated product suppliers
+     * If combinationId is provided, then it only removes product suppliers associated to that combination
+     * If combinationId is null, then it only removes product suppliers that are not associated to any combination
+     *
+     * @param ProductId $productId
+     * @param CombinationId|null $combinationId
+     */
+    public function removeAll(ProductId $productId, ?CombinationId $combinationId = null): void
     {
-        $deletableProductSupplierIds = $this->getDeletableProductSupplierIds($productId->getValue(), $productSuppliers);
+        $product = $this->productRepository->get($productId);
+        $productSuppliersInfo = $this->productSupplierRepository->getProductSuppliersInfo($productId, $combinationId);
+
+        $productSupplierIds = [];
+        foreach ($productSuppliersInfo as $productSupplier) {
+            $productSupplierIds[] = new ProductSupplierId((int) $productSupplier['id_product_supplier']);
+        }
+
+        $this->productSupplierRepository->bulkDelete($productSupplierIds);
+        $this->resetDefaultSupplier($product);
+    }
+
+    /**
+     * @param ProductId $productId
+     * @param array<int, ProductSupplier> $productSuppliers
+     * @param CombinationId|null $combinationId
+     */
+    private function persist(ProductId $productId, array $productSuppliers, ?CombinationId $combinationId = null): void
```

Comment:
we are in ProductSupplierUpdater. To me It sounds exactly what it does :smile: 
`persistSuppliers` doesn't make sense because we are persisting ProductSuppliers :stuck_out_tongue: 
But I suppose you ment persistProductSuppliers. That I can do, altough its abit redundant.